queue_job/controllers/main.py

```diff
@@ -31,6 +31,10 @@ def _try_perform_job(self, env, job):
         job.set_started()
         job.store()
         env.cr.commit()
+        locked = job.lock()
+        if not locked:
+            # When job is considered as 'dead' before actually locking the job
+            raise OperationalError("Trying to lock job that wasn't started")
```

Comment:
Since this is not a postgres error, maybe use another exception type?
Also wondering if it would be slightly easier to read if the raise was in the `lock()` function.

Reply:
A `RetryableJobError` seems more appropriate